homeassistant/components/tibber/sensor.py

```diff
@@ -198,8 +228,8 @@ def unit_of_measurement(self):
         return POWER_WATT
 
     @property
-    def unique_id(self):
-        """Return a unique ID."""
+    def device_id(self):
+        """Return the ID of the physical device this sensor is part of."""
         home = self._tibber_home.info["viewer"]["home"]
         _id = home["meteringPointData"]["consumptionEan"]
         return f"{_id}_rt_consumption"
```

Comment:
This is weird. Device ID needs to be the unique ID of the device. And `unique_id` needs to be the unique ID of the entity, which is part of the device and so the device ID is probably part of the unique ID of the entity. 

Here I would expect `home["meteringPointData"]["consumptionEan"]` to be returned since it's a device ID.

Also, you're removing unique ID here, why?

Reply:
There are two different types of sensors (`TibberSensorElPrice`  and `TibberSensorRT `) and both inherent from `TibberSensor `

> Also, you're removing unique ID here, why?

I moved it to `TibberSensor `


TibberSensorElPrice is a virtual price sensor that is connected to a home
TibberSensorRT is a physical sensor that is connected to a home.

All homes have one (and never more than one) TibberSensorElPrice, but you can have a TibberSensorRT (but not more than one device) if you have an optional physical device.


Today:
TibberSensorElPrice has `home["meteringPointData"]["consumptionEan"]` as `unique_id`
TibberSensorRT has `home["meteringPointData"]["consumptionEan"]_rt_consumption ` as `unique_id`

So `home["meteringPointData"]["consumptionEan"]`  is unique per home, and you can not have more than one TibberSensorElPrice and one TibberSensorRT per home.


So maybe the `device_id` should be defined as `home["meteringPointData"]["consumptionEan"]` for `TibberSensor ` and then the unique ID as `f"{self.device_id}_rt_consumption"` for TibberSensorRT and device id equal unique ID for TibberSensorElPrice? 
Then we also avoid any breaking changes

Appended the changes, so you see what I mean

Reply:
I am confused, as I feel like you're mixing up devices and entity IDs here.

Both entities need to return the same device ID, because they are both related to the home. 

Each entity should have a unique ID. It is not recommended to have 1 entity be the device ID, instead it should be `{device_id}_rt_consumption`, like you have done for 1 entity. The other one should be `{device_id}_el_price` so you don't have a conflicting unique_id schema where 1 entity has a suffix and others have not.

Reply:
Added `_el_price ` now.
I tried to avoid any breaking changes so that why I wanted to have device_id equal enity_id for the price sensor